packages/core/src/render/Frame.tsx

```diff
@@ -44,21 +44,21 @@ export const Frame: React.FC<React.PropsWithChildren<FrameProps>> = ({
     initialData: data || json,
   });
 
-  const isInitialChildrenLoadedRef = useRef(false);
+  const isInitialLoadedRef = useRef(false);
 
-  useEffect(() => {
-    const { initialChildren, initialData } = initialState.current;
+  if (!isInitialLoadedRef.current && initialState.current.initialData) {
+    actions.history.ignore().deserialize(initialState.current.initialData);
+    isInitialLoadedRef.current = true;
+  }
 
-    if (initialData) {
-      actions.history.ignore().deserialize(initialData);
-      return;
-    }
+  useEffect(() => {
+    const { initialChildren } = initialState.current;
```

Comment:
I think you we can simplify this to:
```tsx
  const isLoaded = useRef(false);

  if (!isLoaded.current) {
    const initialData = data || json;

    if (initialData) {
      actions.history.ignore().deserialize(initialData);
    } else {
      const rootNode = React.Children.only(children) as React.ReactElement;

      const node = query.parseReactElement(rootNode).toNodeTree((node, jsx) => {
        if (jsx === rootNode) {
          node.id = ROOT_NODE;
        }
        return node;
      });

      actions.history.ignore().addNodeTree(node);
    }
```

so we can remove the `useEffect` hook completely

Reply:
Thanks! I've applied the changes. This was where I wasn't sure if Creaft.js somehow relied on the children parsing being done after rendering. As far as I can tell looking in more detail - it doesn't. And tests do pass and I couldn't see anything breaking while playing around with the examples in the repo and my own app. So hopefully good to go!

Reply:
Yeap, Craft just parses the children elements if there was no `data` value present - so in this way, the children elements are used to create the default/initial content for the editor